staging/src/k8s.io/kubectl/pkg/cmd/cp/filespec.go

```diff
@@ -0,0 +1,192 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cp
+
+import (
+	"path"
+	"path/filepath"
+	"strings"
+)
+
+type fileSpec struct {
+	PodName      string
+	PodNamespace string
```

Comment:
(optional) If it's not too difficult, it probably makes sense to split this into a remote & local fileSpec, since local files won't have a pod name & namespace. This would make the method signatures for methods like `copyToPod` a bit easier to understand too.

Reply:
That was my initial take into it, but that turned out to unnecessarily complicate the interface and implementation in several places. That's why I went with just the `filePath` bits.